File: app/src/main/java/com/adrcotfas/minibrowser/cache/UrlEntity.java
package com.adrcotfas.minibrowser.cache;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(indices = {@Index(value = {"url"}, unique = true)})
public class UrlEntity {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public @NonNull String url;

    public UrlEntity(int id, @NonNull String url) {
        this.id = id;
        this.url = url;
    }
}

File: app/src/main/java/com/adrcotfas/minibrowser/cache/UrlEntity.kt
package com.adrcotfas.minibrowser.cache

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(indices = [Index(value = ["url"], unique = true)])
class UrlEntity(@field:PrimaryKey(autoGenerate = true) var id: Int, var url: String)